apps/passport-server/src/multiprocess/worker.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * This file runs inside a child process of the zupass server. It comprises
+ * the entire surface area of the tasks that the Zupass server can perform
+ * in the background. You can put tasks that are CPU-bound here, so that they
+ * can be performed in a background thread.
+ *
+ * See {@link MultiProcessService} to understand how exactly this worker is invoked.
+ */
+
+import { SemaphoreSignaturePCDPackage } from "@pcd/semaphore-signature-pcd";
+import { logger } from "../util/logger";
+
+logger("[WORKER] initializing a process. pid: " + process.pid);
+
+const verify = async (serializedSerializedPCD: string): Promise<boolean> => {
+  logger(
+    "[WORKER] verifying a semaphore signature pcd",
+    serializedSerializedPCD
+  );
+
+  const parsed = JSON.parse(serializedSerializedPCD);
+
+  const deserialized = await SemaphoreSignaturePCDPackage.deserialize(
+    parsed.pcd
+  );
+
+  const valid = await SemaphoreSignaturePCDPackage.verify(deserialized);
+
+  logger(`[WORKER] valid: ${valid}`, "pcd:", serializedSerializedPCD);
+
+  return valid;
+};
+
+// It is possible to strongly type the interface between
```

Comment:
What level of strong typing do you have in mind here?  It's a JavaScript library, so I don't think it's very strongly typed.
I do see that it supports multiple named functions, which will be useful when we have multiple use cases.  That's worth saying explicitly in a comment so anyone who looks at this later knows they can add more.

Reply:
Something akin to how we strongly type the API between the Zupass client and the server, but one level simpler.

Reply:
>  I do see that it supports multiple named functions, which will be useful when we have multiple use cases. That's worth saying explicitly in a comment so anyone who looks at this later knows they can add more.

added a comment